java/adapter/avro/src/main/java/org/apache/arrow/consumers/AvroBytesConsumer.java

```diff
@@ -79,4 +79,11 @@ public void setPosition(int index) {
   public FieldVector getVector() {
     return vector;
   }
+
+  @Override
+  public void close() {
+    if (cacheBuffer != null) {
+      cacheBuffer.clear();
```

Comment:
Is this necessary, reading the javado it just seems like this resets the value but doesn't free any resources?  It seems like if anything we should be closing the vectors in the consumers?  But maybe that is a decision for when we support reading only a set number of rows?

Reply:
Thanks, you are right, unnecessary here.
1. close should release vectors but only when exception occurs, otherwise vectors should release by users.
2. the same for iterator API later.
PR updated now.